static/app/utils/useHoverOverlay.tsx

```diff
@@ -102,57 +151,16 @@ function useHoverOverlay(
     containerDisplayMode = 'inline-block',
   }: UseHoverOverlayProps
 ) {
-  const [isVisible, setVisible] = useState(false);
-  const describeById = useMemo(() => domId(`${overlayType}-`), [overlayType]);
   const theme = useTheme();
+  const describeById = useMemo(() => domId(`${overlayType}-`), [overlayType]);
 
-  const isOpen = forceVisible ?? isVisible;
-
+  const [isOpen, setIsOpen] = useState(forceVisible ?? false);
```

Comment:
🤔 doesn't this change the behaviour of `forceVisible`? It's only used as the initial value of the hook; if the user dismisses the overlay `isOpen` will be set to `false` which will hide the overlay even if `forceVisible` is still `true`, no?

Reply:
No, I return `isOpen: forceVisible ?? isOpen` in the end, but I see how that is confusing. I'll find a better way

Reply:
Ah roger, I missed that part

Reply:
Yeah, I dont blame you. I will move it to what it was and use a ref to track the state inside the event hooks so we can avoid the invalidation